api/types/image_summary.go

```diff
@@ -47,6 +46,11 @@ type ImageSummary struct {
 	// Required: true
 	ParentID string `json:"ParentId"`
 
+	// The image platform
+	//
+	// Required: false
+	Platform *ocispec.Platform `json:"Platform,omitempty"`
```

Comment:
We should look if we want the Platform embedded of an explicit sub-struct here.

i.e.; 

```json
{"Name":"myimage","Platform":{"architecture":"arm","os":"linux","variant":"v7"}}
```

or

```json
{"Name":"myimage","architecture":"arm","os":"linux","variant":"v7"}
```

Embedding would also the fields to be accessed both through `image.OS`, `image.Architecture` (etc), as well as `image.Platform`, and in cases where we _only_ need the platform, would allow to unmarshal  directly into a `Platform` (see https://github.com/containerd/containerd/pull/7376)

Reply:
I think the confusing / wrong bit here is that the Image ID (with containerd integration) refers to the digest what we resolved the image to, which is the Manifest _index_ (for multi-arch images), but the `Platform` now only contains a single Platform.

This also introduces the problem shown in your example, where images have the same  name, same ID, but are not unique;

```
REPOSITORY    TAG       IMAGE ID       CREATED      SIZE
..
hello-world   latest    fc6cf906cbfa   4 days ago   20.8kB
hello-world   latest    fc6cf906cbfa   4 days ago   17.5kB
..
```

Reply:
The _id_ issue already exists but only for the containerd integration, we shouldn't talk about this here (I did try though and set the manifest Id but then searching for that manifest directly is not that easy; or at least I didn't want to take too much time doing that right now).

Reply:
We could do embedding indeed, `image inspect` does give a JSON with an embedded platform.

Reply:
How about embedding it in the structure but still keep the `Platform` JSON property?

```golang
// The image platform
//
// Required: false
*ocispec.Platform `json:"Platform,omitempty"`
```